### PR TITLE
feat: enrich and fix time conversion functions

### PR DESCRIPTION
**Pull request by phc1990**
- ✅ Enrich conversion to `datetime` and `Instant` to consider ISO strings
- ✅ Create a `coerce_to_iso` for `datetime`, `Instant` and ISO strings
- ✅ Fix `datetime` to `Instant` conversion where scale was assumed to be in UTC
- ✅ Enriched tests

### CODE REVIEW

File: bindings/python/tools/python/ostk/astrodynamics/converters.py [hunk 1]
```diff
@@ -46,7 +55,41 @@ def coerce_to_instant(value: Instant | datetime) -> Instant:
     if isinstance(value, Instant):
         return value
 
-    return Instant.date_time(value, Scale.UTC)
```
Comment (phc1990) [inline bindings/python/tools/python/ostk/astrodynamics/converters.py:49]: This was assuming that `value` was in UTC, see line below for the fix.

File: bindings/python/test/test_converters.py [hunk 2]
```diff
@@ -25,24 +28,167 @@
...
+    value = datetime(2020, 1, 2, 3, 4, 5, 123456, tzinfo=timezone.utc)
     assert coerce_to_datetime(value) == value
 
+    value = datetime(
+        2020, 1, 2, 3, 4, 5, 123456, tzinfo=timezone(timedelta(seconds=3600))
+    )
+    assert coerce_to_datetime(value) == value
+
+
+def test_coerce_to_datetime_success_iso():
```
Comment (vishwaLoft) [inline bindings/python/test/test_converters.py:52]: Maybe add a test for coercing "2020-01-02T03:04:05.123456Z"
And also without milliseconds/microseconds?

File: bindings/python/tools/python/ostk/astrodynamics/converters.py [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Apache License 2.0
 
+from typing import Union
```
Comment (vishwaLoft) [inline bindings/python/tools/python/ostk/astrodynamics/converters.py:3]: Why is this necessary? Can't we use |

File: bindings/python/tools/python/ostk/astrodynamics/converters.py [hunk 4]
```diff
@@ -15,12 +17,12 @@
 from ostk.physics.coordinate import Frame
 
 
-def coerce_to_datetime(value: Instant | datetime) -> datetime:
+def coerce_to_datetime(value: Union[Instant, datetime, str]) -> datetime:
```
Comment (vishwaLoft) [inline bindings/python/tools/python/ostk/astrodynamics/converters.py:20]:
```suggestion
def coerce_to_datetime(value: Instant | datetime | str) -> datetime:
```

File: bindings/python/tools/python/ostk/astrodynamics/converters.py [hunk 5]
```diff
@@ -46,7 +55,41 @@ def coerce_to_instant(value: Instant | datetime) -> Instant:
...
+
+def coerce_to_iso(
+    value: Union[Instant, datetime, str], timespec: str = "microseconds"
+) -> Instant:
+    """
+    Return an ISO string from value.
+
+    Args:
+        value (Instant | datetime | str): A value to coerce.
+        timespec (str): A time resolution ("microseconds" by default).
```
Comment (vishwaLoft) [inline bindings/python/tools/python/ostk/astrodynamics/converters.py:76]:
```suggestion
        timespec (str): A time resolution. Defaults to "microseconds".
```
Nit.

File: bindings/python/tools/python/ostk/astrodynamics/converters.py [hunk 6]
```diff
@@ -46,7 +55,41 @@ def coerce_to_instant(value: Instant | datetime) -> Instant:
...
+        return coerce_to_iso(coerce_to_datetime(value), timespec=timespec)
+
+    elif isinstance(value, datetime):
+        return value.isoformat(timespec=timespec)
+
+    elif isinstance(value, Instant):
+        return coerce_to_iso(coerce_to_datetime(value), timespec=timespec)
+
+    else:
+        raise TypeError("Argument must be a datetime, an Instant, or a str.")
```
Comment (vishwaLoft) [inline bindings/python/tools/python/ostk/astrodynamics/converters.py:92]:
```suggestion
    if isinstance(value, str):
        return coerce_to_iso(coerce_to_datetime(value), timespec=timespec)

    if isinstance(value, datetime):
        return value.isoformat(timespec=timespec)

    if isinstance(value, Instant):
        return coerce_to_iso(coerce_to_datetime(value), timespec=timespec)

    raise TypeError("Argument must be a datetime, an Instant, or a str.")
```
Nit

File: bindings/python/tools/python/ostk/astrodynamics/converters.py [hunk 7]
```diff
@@ -46,7 +55,41 @@ def coerce_to_instant(value: Instant | datetime) -> Instant:
...
 
-    return Instant.date_time(value, Scale.UTC)
+    elif isinstance(value, datetime):
+        return Instant.date_time(value.astimezone(tz=timezone.utc), Scale.UTC)
+
+    elif isinstance(value, str):
+        return coerce_to_instant(coerce_to_datetime(value))
+
+    else:
+        raise TypeError("Argument must be a datetime, an Instant, or a str.")
```
Comment (vishwaLoft) [inline bindings/python/tools/python/ostk/astrodynamics/converters.py:65]:
```suggestion
    if isinstance(value, datetime):
        return Instant.date_time(value.astimezone(tz=timezone.utc), Scale.UTC)

    if isinstance(value, str):
        return coerce_to_instant(coerce_to_datetime(value))

    raise TypeError("Argument must be a datetime, an Instant, or a str.")
```

**Review by vishwaLoft — COMMENTED** [review]
Looks great man!

File: bindings/python/test/test_converters.py (hunk 2 above)
Comment (phc1990) [reply bindings/python/test/test_converters.py:52]: Added Z format with and without decimals for all 3 functions.

File: bindings/python/tools/python/ostk/astrodynamics/converters.py (hunk 3 above)
Comment (phc1990) [reply bindings/python/tools/python/ostk/astrodynamics/converters.py:3]: I was just using `|` but the linter didn't like it.
Seems ok now though, so adapting 👍

**Review by vishwa2710 — APPROVED** [review]

**Comment by Derollez** [comment]
Thanks for aligning 🙏
